hbase-client/src/main/java/org/apache/hadoop/hbase/client/MasterRegistry.java

```diff
@@ -89,11 +97,17 @@
   private final int hedgedReadFanOut;
 
   // Configured list of masters to probe the meta information from.
-  private final ImmutableMap<ServerName, ClientMetaService.Interface> masterAddr2Stub;
+  private volatile ImmutableMap<ServerName, ClientMetaService.Interface> masterAddr2Stub;
 
   // RPC client used to talk to the masters.
   private final RpcClient rpcClient;
   private final RpcControllerFactory rpcControllerFactory;
+  private final int rpcTimeoutMs;
+  // For synchronizing on refreshing the master end-points
+  private final Object refreshMasters = new Object();
```

Comment:
nit: `static final` here? Anyways, `MasterRegistry` is singleton right? (if not by design, but by usage)

Reply:
Oops, yes. This better be just final.